src/zoid/buttons/util.js

```diff
@@ -115,6 +115,54 @@ export function getVenmoExperiment(experiment : ?Experiment) : VenmoExperiment {
     }
 }
 
+export function createNoPaylaterExperiment() : Experiment | void {
+    return inlineMemoize(createNoPaylaterExperiment, () => {
+        const disableFunding = getDisableFunding();
+        const isDisableFundingPaylater = disableFunding && disableFunding.indexOf(FUNDING.PAYLATER) !== -1;
+        const enableFunding = getEnableFunding();
+        const isEnableFundingPaylater = enableFunding && enableFunding.indexOf(FUNDING.PAYLATER) !== -1;
+
+        const fundingEligibility = getFundingEligibility();
+        const isEligibleForPaylater = fundingEligibility && fundingEligibility[FUNDING.PAYLATER] && fundingEligibility[FUNDING.PAYLATER].eligible;
+
+        if (isDevice()) {
+            // No experiment because ineligible, already forced on or off, unsupported browser
+            if (!isEligibleForPaylater
```

Comment:
I think this is probably a better example where the function calls might help cleanup the traceabiltiy aspect of what a 'complex' conditional is testing for.

Reply:
I agree that it isn't pretty, but it follows the convention set by the venmoExperiment. I don't think that creating a function is really necessary tbh

Reply:
I think these eligibility checks need to apply to both conditions (isDevice() is true or false). Venmo has some unique rules for desktop eligibility that we should not apply here.